CLDR-18938 Add message format to spec modifications (#5088)
unicode-org/cldr: docs/ldml/tr35-modifications.md

The rules have been converted to a “flat” format, which is easier for clients to handle (the old format will be retained for one more release).

### Units of Measurement
* [Unit Syntax](tr35-general.md#unit-syntax) Simplified the EBNF `product_unit`, and added an additional well-formedness constraint for mixed units.
* [Unit Identifier Normalization](#tr35-general.md) Modified the normalization process
* [Mixed Units](tr35-general.md#mixed-units) Modified the guidance for handling precision.

### Collation
* [Collation](tr35-collation.md) Added the new `FractionalUCA_blanked.txt` to the root collation data files.

### MessageFormat
#### Normative changes to Stable parts of the specification:

> * Syntax and data model errors are now required to be prioritized over other errors ([\#1011](https://github.com/unicode-org/message-format-wg/pull/1011))  

Reviewer: ```suggestion
* Syntax and data model errors now must be prioritized over other errors ([\#1011](https://github.com/unicode-org/message-format-wg/pull/1011))  
```
